### ISSUE TITLE
Web UI not displaying correctly on reverse proxy

### ISSUE THREAD

**Issue by kyrypto** [Haehnchen/crypto-trading-bot#95]
![Screenshot 2019-12-21 at 1 11 54 PM](https://user-images.githubusercontent.com/38748393/71312706-8befe000-23f3-11ea-803b-e57837691fac.png)
Everything seems to work besides that it just doesn't look right.
Images are not displaying, including the logo. 
I'm not getting any errors however. Note I am using Nginx reverse proxy on a VPS, unsure if that has anything to do with it.

**Comment by kyrypto** [comment]
After a LOT of research I found a solution to this. It is the reverse proxy, but the solution is not with NGINX, you have to switch to Apache. Here is my working config, including SSL and password authentication:
Not you have to enable the rewrite module, proxy/proxy_http modules, and SSL modules.  Create a password with: " htpasswd -c /etc/apache2/.htpasswd user1" 

With a quick Google search you can find the relevant modules to enable 


Please add this to the documentation

<VirtualHost *:443>
    ServerAdmin webmaster@example.com
    DocumentRoot "/home/user/http/example.com"
    ServerName example.com:443
    ServerAlias example.com:443
ProxyPreserveHost On
ProxyPass "/"  "http://127.0.0.1:8080/"
ProxyPassReverse "/"  "http://12.0.0.1:8080/"
RewriteEngine on
RewriteRule ^/(.*)/(images|javascripts|stylesheets)(.*) /$1/$2$3
<Proxy *>
        Order deny,allow
        Allow from all
        Authtype Basic
        Authname "Password Required"
        AuthUserFile /etc/apache2/.htpasswd
        Require valid-user
    </Proxy>
    SSLEngine on
    ErrorLog "/var/log/httpd/example.com-error_log"
    CustomLog "/var/log/httpd/example.com-access_log"common

    <Directory "/home/user/http/example.com">
        Require all granted
    </Directory>
        ServerAlias example.com
SSLCertificateFile /etc/letsencrypt/live/example.com/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/example.com/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>

**Comment by hatzkel** [comment]
This is a problem I think you alone experienced. I tried to look into it but never responded last night because I could not reproduce. I figure it had something to do with your configuration.... Why don't you just run Docker? You are causing yourself unnecessary headaches by trying to have the development environment on you host when you could just run the docker container on any host and have all the dependencies and necessary configuration right there? There is no reason for you to have to run apache over nginx2 when you simply could have configured nginx correctly...anyone coming across this is just going to be confused.

**Comment by hatzkel** [comment]
This is a problem I think you alone experienced. I tried to look into it but never responded last night because I could not reproduce. I figure it had something to do with your configuration.... Why don't you just run Docker? You are causing yourself unnecessary headaches by trying to have the development environment on your host when you could just run the docker container on any host and have all the dependencies and necessary configuration right there? There is no reason for you to have to run apache over nginx2 when you simply could have configured nginx correctly...anyone coming across this is just going to be confused.

**Comment by kyrypto** [comment]
> This is a problem I think you alone experienced. I tried to look into it but never responded last night because I could not reproduce. I figure it had something to do with your configuration.... Why don't you just run Docker? You are causing yourself unnecessary headaches by trying to have the development environment on you host when you could just run the docker container on any host and have all the dependencies and necessary configuration right there? There is no reason for you to have to run apache over nginx2 when you simply could have configured nginx correctly...anyone coming across this is just going to be confused.

I do run within Docker, this problem only occurs within reverse proxy, the solution to this issue was Apache, so if you have the correct config for Nginx reverse proxy please share.

**Comment by hatzkel** [comment]
Now this is something we are going to have to fix before ever going production since how nice it would be to put on a eu-west-1 aws free tier in the same building as bitmex with a 0.352 ms ping. nginx has a rewrite module have you tried this and like settings to which you applied?

**Comment by kyrypto** [comment]
> Now this is something we are going to have to fix before ever going production since how nice it would be to put on a eu-west-1 aws free tier in the same building as bitmex with a 0.352 ms ping. nginx has a rewrite module have you tried this and like settings to which you applied?

I had to dig online to find a solution, I came up with squat for Nginx, but for Apache there was more documentation. If someone knows how to use that module for a fix it would be great.

**Comment by hatzkel** [comment]
https://www.nginx.com/blog/creating-nginx-rewrite-rules/

Might be a good start. I would try writing your apache rules but in a way nginx understands it and go from there. No need to start changing the stack.... yet.

**Comment by hatzkel** [comment]
https://www.nginx.com/blog/creating-nginx-rewrite-rules/

Might be a good start. I would try writing your apache rules into nginx format first.

**Comment by kyrypto** [comment]
> https://www.nginx.com/blog/creating-nginx-rewrite-rules/
> 
> Might be a good start. I would try writing your apache rules into nginx format first.

I'll see what I can do, this would be the setting to convert: RewriteRule ^/(.)/(images|javascripts|stylesheets)(.*) /$1/$2$3

**Comment by kyrypto** [comment]
This indeed seemed to be an issue with my own Nginx config after some more troubleshooting. I used a config generator, and it broke the setup. So for others using nginxconfig.io - save a headache and delete the following in your server config:
include nginxconfig.io/general.conf;
}

However the issue did occur on Apache when I tried it, and I found the fix for it so it'll hopefully help someone who prefers Apache. I'll close this now thanks dev for the hard work.
